### PR TITLE
Admin Panel Forms validation & markup cleanup

### PR DESCRIPTION
**Pull request by damianlegawiec**
This PR:
- unifies validation error handling across all Admin Panel forms
- adds missing inline validation errors to Admin Panel forms  (via `error_message_on`)
- fixes double validation errors - validation errors were rendered via `admin/shared/error_messages` partial and flashes, which wasn't a good UX solution
- unifies Admin Panel forms with the usage of `field_container` method
- fixes double `form-group` divs markup errors

### CODE REVIEW

File: backend/app/views/spree/admin/tax_rates/_form.html.erb [hunk 1]
```diff
@@ -13,10 +13,13 @@
...
-          <%= f.label :amount, Spree.t(:rate) %>
-          <%= f.text_field :amount, :class => 'form-control' %>
+          <%= f.field_container :amount, class: ["form-group"] do %>
+            <%= f.label :amount, Spree.t(:rate) %>
+            <%= f.text_field :amount, :class => 'form-control' %>
+            <%= f.error_message_on :amount %>
+          <% end %>
           <p class="help-block">
-            <%= Spree.t(:tax_rate_amount_explanation) %>
+          <%= Spree.t(:tax_rate_amount_explanation) %>
```
Comment (Mafi88) [inline backend/app/views/spree/admin/tax_rates/_form.html.erb]: I think that indentation was ok here.

**Comment by Mafi88** [comment]
Otherwise :+1:

File: backend/app/views/spree/admin/roles/edit.html.erb [hunk 2]
```diff
@@ -6,6 +6,8 @@
   <%= back_to_list_button(Spree::Role.model_name.human, admin_roles_path) %>
 <% end %>
 
+<%= render :partial => 'spree/admin/shared/error_messages', :locals => { :target => @role } %>
```
Comment (priyank-gupta) [inline backend/app/views/spree/admin/roles/edit.html.erb]: @damianlegawiec We should use new ruby syntax for any new code.

Comment (damianlegawiec) [inline backend/app/views/spree/admin/roles/edit.html.erb]: @priyank-gupta this is @vinayvinsol commit so if I'll fix it now it's still will change the history, same as doing `rubocop -a` later on

Comment (priyank-gupta) [inline backend/app/views/spree/admin/roles/edit.html.erb]: @damianlegawiec OK. Lets leave it for now and change hash syntax as per new ruby in another PR.

**Pull request merged**